frontend/src/components/MapView/DateSelector/TimelineItems/TimelineItem/index.tsx

```diff
@@ -0,0 +1,93 @@
+import { WithStyles, createStyles, withStyles } from '@material-ui/core';
+import React, { memo, useCallback } from 'react';
+import 'react-datepicker/dist/react-datepicker.css';
+import { DateItem, DateRangeType } from '../../../../../config/types';
+
+const TimelineItem = memo(
+  ({
+    classes,
+    concatenatedLayers,
+    currentDate,
+    index,
+    clickDate,
+    dateItemStyling,
+  }: TimelineItemProps) => {
+    const handleClick = useCallback(
+      (dateIndex: number) => {
+        return () => {
+          clickDate(dateIndex);
+        };
+      },
+      [clickDate],
+    );
+
+    return (
+      <>
+        {concatenatedLayers.map(
+          (layerDates: DateItem[], layerIndex: number) => {
+            // TODO: fix not really efficient algorithm
+            const matchingDateItemInLayer = layerDates.find(
+              f =>
+                new Date(f.displayDate).toDateString() ===
```

Comment:
Let's merge https://github.com/WFP-VAM/prism-app/pull/866 first and use this util

Reply:
Done !